project-koku/koku
# Sources: authentication.create

- authentication.create will check if resource_id is associated with a source that is associated with cost-management. If not, ignore the message.

testing:
POST authentication not associated with cost-management:
```
[2020-03-04 23:02:33,612] INFO None Cost Management Message to process: {'event_type': 'Authentication.create', 'offset': 89, 'resource_id': 8, 'auth_header': 'eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTAwMDEiLCAidHlwZSI6ICJVc2VyIiwgInVzZXIiOiB7InVzZXJuYW1lIjogInVzZXJfZGV2IiwgImVtYWlsIjogInVzZXJfZGV2QGZvby5jb20iLCAiaXNfb3JnX2FkbWluIjogdHJ1ZX19LCAiZW50aXRsZW1lbnRzIjogeyJjb3N0X21hbmFnZW1lbnQiOiB7ImlzX2VudGl0bGVkIjogdHJ1ZX19fQ=='}
[2020-03-04 23:02:33,614] INFO None Processing Event: {'event_type': 'Authentication.create', 'offset': 89, 'resource_id': 8, 'auth_header': 'eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTAwMDEiLCAidHlwZSI6ICJVc2VyIiwgInVzZXIiOiB7InVzZXJuYW1lIjogInVzZXJfZGV2IiwgImVtYWlsIjogInVzZXJfZGV2QGZvby5jb20iLCAiaXNfb3JnX2FkbWluIjogdHJ1ZX19LCAiZW50aXRsZW1lbnRzIjogeyJjb3N0X21hbmFnZW1lbnQiOiB7ImlzX2VudGl0bGVkIjogdHJ1ZX19fQ=='}
[2020-03-04 23:02:33,788] INFO None Resource id 8 not associated with cost-management.
```

1. Create source by posting to sources-api in the following order: 
sources -> endpoints -> application -> authentication and PATCH to koku
See successful source creation
2. Create source by posting to sources-api in the following order:
sources -> endpoints -> authentication -> application and PATCH to koku
See successful source creation

With this PR, we _should_ be able to create a source in CI, un-assign cost-management and then re-assign cost-management without failures.
